L105-python-expose-new-error-types.md

```diff
@@ -0,0 +1,30 @@
+L105: Python Add New Error Types
+----
+* Author(s): XuanWang-Amos
+* Approver: gnossen
+* Status: In Review
+* Implemented in: Python
+* Last updated: 08/21/2023
+* Discussion at: TODO
+
+## Abstract
+
+Add two new errors in grpc public API:
+  * BaseError
+  * AbortError
+
+## Background
+
+In case of abort, currently we don't log anything, exposing those error types allows user to catch and handle aborts if they want.
+
+## Proposal
+
+Add AbortError and BaseError in public API.
+
+## Rationale
+
+The Async API [has similar errors](https://github.com/grpc/grpc/blob/v1.57.x/src/python/grpcio/grpc/aio/__init__.py#L23,L24). We're refactoring code so those errors will also be used in Sync API. Adding them to the Sync API will help us keep the two stacks in sync and allow users of the Sync implementation to catch and handle aborts.
+
```

Comment:
We probably want to mention that adding this will allow users to type:

```python
try:
  do_grpc_stuff()
except grpc.BaseError as e:
  # handle error
```

and be confident that they're catching _all_ gRPC exceptions.

Reply:
We also have `RpcError` exposed to user, I think we should mention their differences, when should user expect RpcError?

Reply:
If RpcError is not currently a subclass of BaseError, I think we should make it a subclass of BaseError. Ditto for all other exception types. The ultimate goal is to enable the snippet above.

Reply:
Make sense, I'll change `RpcError` to be a subclass of `BaseError` and add it to this gRFC.